x/sequencer/keeper/msg_server_create_sequencer.go

```diff
@@ -50,7 +50,13 @@ func (k msgServer) CreateSequencer(goCtx context.Context, msg *types.MsgCreateSe
 		if !isInitialOrAllAllowed {
 			return nil, types.ErrNotInitialSequencer
 		}
-		if err := k.rollappKeeper.SealRollapp(ctx, msg.RollappId); err != nil {
+
+		// check pre launch time
+		if rollapp.PreLaunchTime.After(ctx.BlockTime()) {
+			return nil, types.ErrBeforePreLaunchTime
```

Comment:
if it's not set, the expression is always false

Reply:
I see. imo this implicitness is confusing. we can add a comment to clarify or alternatively explictly check this in the case this sequencer is launched for rollapp with IRO. 

also thought of suggesting  it would be cleaner to move `preLaunchTime` to only live on the IRO object if it's only IRO related but than checked the IRO object and I see it also has a `preLaunchTime`. and seems like it's a duplicate var living in both places. so I think worth just leaving it on the IRO, change it's name to `EndTime` and make sure a rollapp can't start until `rollapp.iro.EndTime` is over. 

would also address https://github.com/dymensionxyz/dymension/pull/1201#discussion_r1751039489

Reply:
actually I think it's better to have `preLaunchTime` as rollapp's field, as it might be enforced for different reasons in the future, not only from IRO.
I left it on the IRO plan object as well for clearity, but it's not used and can be removed from the plan store if u prefer.

Reply:
> actually I think it's better to have preLaunchTime as rollapp's field, as it might be enforced for different reasons in the future, not only from IRO.

I'm fine with that.

> I left it on the IRO plan object as well for clearity, but it's not used and can be removed from the plan store if u prefer.

also fine with that, but imo we need to change it's name to `EndTime` cause I don't see the logic of `Pre-Launch time` on the IRO object, especially with it's comment related to rollapp.